src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -283,11 +283,20 @@ private static void OnDocEnded(C4Replicator* repl, bool pushing, IntPtr numDocs,
                 return;
             }
 
+            var conflictTasks = new ConcurrentBag<Task>();
+            var replicatedDocuments = new ConcurrentBag<ReplicatedDocument>();
             var documentReplications = new List<ReplicatedDocument>((int)numDocs);
             for (int i = 0; i < (int) numDocs; i++) {
                 var current = docs[i];
-                documentReplications.Add(new ReplicatedDocument(current->docID.CreateString() ?? "", 
-                    current->flags, current->error, current->errorIsTransient));
+                if (!pushing && current->error.domain == C4ErrorDomain.LiteCoreDomain &&
+                    current->error.code == (int)C4ErrorCode.Conflict) {
+                    conflictTasks.Add(Task.Run(() =>
+                        replicatedDocuments.Add(new ReplicatedDocument(current->docID.CreateString() ?? "",
+                            current->flags, current->error, current->errorIsTransient))));
```

Comment:
There is no reason for the above to be asynchronous.  It should just be a simple add to a collection.